components/button/EffectButton.tsx

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+
+const defaultClass = "flex p-2 rounded items-center";
+const effectClass = "color-effect";
+
+interface Props {
+  children: React.ReactNode;
+  value: string;
+  [key: string]: any;
+}
+
+export const EffectButton = React.forwardRef<HTMLDivElement, Props>(function EffectButton(props: Props, ref) {
+  const [className, setClassName] = React.useState<string>(defaultClass);
+  const [currentValue, setCurrentValue] = React.useState<string>(props.value);
+
+  const { children, value, ...rest } = props;
+
+  React.useEffect(() => {
+    if (props.value !== currentValue) {
+      {
+        setClassName(addClassName(className));
+        setCurrentValue(props.value);
+      }
+
+      const timeout = setTimeout(() => {
+        setClassName(defaultClass);
+        clearTimeout(timeout);
+      }, 1000)
```

Comment:
So this is completely retarded Javascript bullshit, but I think this is how it works and I couldn't find any other way to do it. To me it looks like the timeout instances are cleaned up properly, but what do I know lol.